### PR TITLE
Update dependencies, add error-handling

### PR DESCRIPTION
**Pull request by mjzffr**
- Updates to mozlog and treeherder-client.
  *\* The mozlog change might break your parsing of the steeplechase logs -- I'll do a test run in pf-jenkins once the steeplechase machines are back online
- Added try blocks around Treeherder submission code

### CODE REVIEW

File: treeherding.py [hunk 1]
```diff
@@ -78,7 +77,7 @@
             'platform': 'osx-10-6',
             'architecture': 'x86_64',
         }
-    }, # ** Windows
+    },  # ** Windows
```
Comment (sydvicious) [inline treeherding.py:80]: Why the extra space?

**Comment by sydvicious** [comment]
One nit question, but basically looks good.

File: treeherding.py (hunk 1 above)
Comment (mjzffr) [inline treeherding.py:80]: flake8 was nagging me about it

**Comment by sydvicious** [comment]
Question answered. Looks good! Thanks.

**Comment by mjzffr** [comment]
Great. I'll flag you when I get results from a few test runs.

**Pull request merged**